enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1220 @@
+---
+title: change-management-and-maintenance-schedules
+authors:
+  - @jupierce
+reviewers: 
+  - TBD
+approvers: 
+  - @sdodson
+  - @jharrington22
+api-approvers:
+  - TBD    
+creation-date: 2024-02-29
+last-updated: 2024-02-29
+
+tracking-link:
+  - TBD
+  
+---
+
+# Change Management and Maintenance Schedules
+
+## Summary
+Implement high level APIs for change management which allow
+standalone and Hosted Control Plane (HCP) clusters a measure of configurable control
+over when control-plane or worker-node configuration rollouts are initiated. 
+As a primary mode of configuring change management, implement an option
+called Maintenance Schedules which define reoccurring windows of time (and specifically
+excluded times) in which potentially disruptive changes in configuration can be initiated. 
+
+Material changes not permitted by change management configuration are left in a 
+pending state until such time as they are permitted by the configuration. 
+
+Change management enforcement _does not_ guarantee that all initiated
+material changes are completed by the close of a permitted change window (e.g. a worker-node
+may still be draining or rebooting) at the close of a maintenance schedule, 
+but it does prevent _additional_ material changes from being initiated. 
+
+Change management enforcement _does not_ attempt to define or control the detailed state of the
+system. It only pertains to whether controllers which support change management 
+will attempt to initiate material change themselves. For example, if changes are paused in the middle
+of a cluster update and a node is manually rebooted, change management does not define
+whether the node will rejoin the cluster with the new or old version. 
+
+A "material change" may vary by cluster profile and subsystem. For example, a 
+control-plane update (all components and control-plane nodes updated) is implemented as
+a single material change (e.g. the close of a scheduled permissive window
+will not suspend its progress). In contrast, the rollout of worker-node updates is
+more granular (you can consider it as many individual material changes) and  
+the end of a permitted change window will prevent additional worker-node updates 
+from being initiated.

```

Comment:
How will we define and track (long-term) what is a material change for a given profile and subsystem? It seems hard to rely just on intuition, how will we avoid bugs like "cluster operator $late started updating after permissive window closed, why?".

Reply:
Thankfully there are only two subsystems right now: worker-nodes and control-plane. The difference between the two is that a control-plane update is treated as a single material change and worker-node updates are treated as one per node. In short, control-plane updates can't be paused in the middle at the end of a permissive window. 
API docs and user docs will need to clearly communicate expectations for anything that supports the stanza. I agree that relying solely on intuition could lead to a bad time.